### ISSUE TITLE
IntObjectOpenHashMap.equals() returns true for maps with differing key sets [HPPC-37]

### ISSUE THREAD

**Issue by dweiss** [carrotsearch/hppc#80]
Example test code:

```Java
IntObjectOpenHashMap<String> m1 = new IntObjectOpenHashMap<String>();
m1.put(2, "x");
IntObjectOpenHashMap<String> m2 = new IntObjectOpenHashMap<String>();
m2.put(3, "x");
m1.equals(m2); // returns true
m2.equals(m1); // returns true
```

---
Issue: HPPC-37 (migrated from JIRA), created by Stanisław Osiński (@stanislawosinski), resolved Dec 15 2010

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

This bug would manifest itself iff:
- the two compared maps had identical key set sizes,
- the two compared maps had different key sets; values of keys shared by both maps are still compared for equality.

**Comment by dweiss** [comment]
Comment by Dawid Weiss (@dweiss) (migrated from JIRA)

Fixed in trunk and on 0.3.2 tag-branch.
